drivers/sensor/sm351lt/sm351lt.c

```diff
@@ -0,0 +1,271 @@
+	struct sm351lt_data *data = dev->data;
+
+	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_PROX) {
+		return -ENOTSUP;
+	}
+
+	data->sample_status = gpio_pin_get(data->bus, config->gpio_pin);
+
+	return 0;
+}
+
+static int sm351lt_channel_get(struct device *dev,
+			       enum sensor_channel chan,
+			       struct sensor_value *val)
+{
+	struct sm351lt_data *data = dev->data;
+
+	if (chan == SENSOR_CHAN_PROX) {
+		val->val1 = data->sample_status;
+		val->val2 = 0;
+	} else {
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+#if CONFIG_SM351LT_TRIGGER
+static int sm351lt_attr_set(struct device *dev, enum sensor_channel chan,
+			    enum sensor_attribute attr,
+			    const struct sensor_value *val)
+{
+	struct sm351lt_data *data = dev->data;
+
+	if (chan == SENSOR_CHAN_PROX) {
+		if (attr == SENSOR_ATTR_SM351LT_TRIGGER_TYPE) {
+			/* Interrupt triggering type */
+			data->trigger_type = val->val1;
+		} else {
+			return -ENOTSUP;
+		}
+	} else {
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int sm351lt_attr_get(struct device *dev, enum sensor_channel chan,
+			    enum sensor_attribute attr,
+			    struct sensor_value *val)
+{
+	struct sm351lt_data *data = dev->data;
+
+	if (chan == SENSOR_CHAN_PROX) {
+		if (attr == SENSOR_ATTR_SM351LT_TRIGGER_TYPE) {
+			/* Interrupt triggering type */
+			val->val1 = data->trigger_type;
+			val->val2 = 0;
+		} else {
+			return -ENOTSUP;
+		}
+	} else {
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+#endif
+
+static const struct sensor_driver_api sm351lt_api_funcs = {
+	.sample_fetch = sm351lt_sample_fetch,
+	.channel_get = sm351lt_channel_get,
+#if CONFIG_SM351LT_TRIGGER
+	.attr_set = sm351lt_attr_set,
+	.attr_get = sm351lt_attr_get,
+	.trigger_set = sm351lt_trigger_set,
+#endif
+};
+
+static int sm351lt_init(struct device *dev)
+{
+	const struct sm351lt_config *const config = dev->config;
+	struct sm351lt_data *data = dev->data;
+	uint32_t ret;
+
+	data->bus = device_get_binding(config->bus_name);
+	if (!data->bus) {
+		LOG_ERR("gpio bus not found: %s", config->bus_name);
+		return -ENODEV;
+	}
+
+	ret = gpio_pin_configure(data->bus, config->gpio_pin,
+				 (config->gpio_flags | GPIO_INPUT));
+	if (ret) {
+		LOG_ERR("failed to configure gpio: %d", ret);
+		return ret;
+	}
+
+#if defined(CONFIG_SM351LT_TRIGGER)
+#if defined(CONFIG_SM351LT_TRIGGER_OWN_THREAD)
+	k_sem_init(&data->gpio_sem, 0, UINT_MAX);
+
+	k_thread_create(&data->thread, data->thread_stack,
+			CONFIG_SM351LT_THREAD_STACK_SIZE,
+			(k_thread_entry_t)sm351lt_thread, dev, NULL,
+			NULL, K_PRIO_COOP(CONFIG_SM351LT_THREAD_PRIORITY),
+			0, K_NO_WAIT);
+
+#if defined(CONFIG_THREAD_NAME) && defined(CONFIG_THREAD_MAX_NAME_LEN)
+	/* Sets up thread name based on the device name */
+	char thread_name[CONFIG_THREAD_MAX_NAME_LEN];
+	memset(thread_name, 0, CONFIG_THREAD_MAX_NAME_LEN);
+	snprintf(thread_name, CONFIG_THREAD_MAX_NAME_LEN, "sen_%s", dev->name);
+	k_thread_name_set(&data->thread, thread_name);
```

Comment:
Why not just use the dev name here and skip the runtime string manipulation?

Reply:
For debugging purposes, it prefixes the thread name with 'sen_' to indicate sensor